Userland/Libraries/LibJS/Runtime/PropertyName.h

```diff
@@ -48,7 +48,8 @@ class PropertyName {
         VERIFY(index >= 0);
     }
 
-    PropertyName(char const* chars)
+    template<typename T = char const*>
+    PropertyName(char const* chars) requires(!IsNullPointer<T>)
```

Comment:
errr...what's this do, exactly?
it seems to always have a defaulted template parameter of type `char const*`, you're not using T for the parameter type, so it doesn't really...do anything AFAICT

Reply:
Without:

```
../../Userland/Libraries/LibJS/Runtime/ArrayIteratorPrototype.cpp: In static member function ‘static JS::Value JS::ArrayIteratorPrototype::next(JS::VM&, JS::GlobalObject&)’:
../../Userland/Libraries/LibJS/Runtime/ArrayIteratorPrototype.cpp:73:34: error: conversion from ‘int’ to ‘const JS::PropertyName’ is ambiguous
   73 |     entry_array->define_property(0, Value(static_cast<i32>(index)));
      |                                  ^
In file included from ../../Userland/Libraries/LibJS/Runtime/Object.h:17,
                 from ../../Userland/Libraries/LibJS/Runtime/Array.h:9,
                 from ../../Userland/Libraries/LibJS/Runtime/ArrayIteratorPrototype.cpp:7:
../../Userland/Libraries/LibJS/Runtime/PropertyName.h:50:5: note: candidate: ‘JS::PropertyName::PropertyName(const char*)’
   50 |     PropertyName(char const* chars)
      |     ^~~~~~~~~~~~
../../Userland/Libraries/LibJS/Runtime/PropertyName.h:44:5: note: candidate: ‘JS::PropertyName::PropertyName(u32)’
   44 |     PropertyName(u32 index)
      |     ^~~~~~~~~~~~
In file included from ../../Userland/Libraries/LibJS/Runtime/Array.h:9,
                 from ../../Userland/Libraries/LibJS/Runtime/ArrayIteratorPrototype.cpp:7:
../../Userland/Libraries/LibJS/Runtime/Object.h:92:26: note:   initializing argument 1 of ‘bool JS::Object::define_property(const JS::PropertyName&, JS::Value, JS::PropertyAttributes, bool)’
   92 |     bool define_property(const PropertyName&, Value value, PropertyAttributes attributes = default_attributes, bool throw_exceptions = true);
      |                          ^~~~~~~~~~~~~~~~~~~
```

so that's definitely doing more than nothing :)

Reply:
oh, it just can't deduce `T`, so it ignores this declaration altogether
you might as well remove it :P

Reply:
Removed:

```
In file included from ../../Tests/LibJS/test-js.cpp:8:
../../Userland/Libraries/LibTest/JavaScriptTestRunner.h: In member function ‘virtual void Test::JS::TestRunnerGlobalObject::initialize_global_object()’:
../../Userland/Libraries/LibTest/JavaScriptTestRunner.h:210:21: error: invalid user-defined conversion from ‘const char [7]’ to ‘const JS::PropertyName&’ [-fpermissive]
  210 |     define_property("global", this, JS::Attribute::Enumerable);
      |                     ^~~~~~~~
In file included from ../../Userland/Libraries/LibJS/Runtime/Object.h:17,
                 from ../../Userland/Libraries/LibJS/Runtime/EnvironmentRecord.h:9,
                 from ../../Userland/Libraries/LibJS/Bytecode/Op.h:17,
                 from ../../Userland/Libraries/LibJS/Bytecode/Generator.h:14,
                 from ../../Userland/Libraries/LibJS/Bytecode/Interpreter.h:9,
                 from ../../Userland/Libraries/LibTest/JavaScriptTestRunner.h:22,
                 from ../../Tests/LibJS/test-js.cpp:8:
../../Userland/Libraries/LibJS/Runtime/PropertyName.h:44:5: note: candidate is: ‘JS::PropertyName::PropertyName(u32)’ (near match)
   44 |     PropertyName(u32 index)
      |     ^~~~~~~~~~~~
../../Userland/Libraries/LibJS/Runtime/PropertyName.h:44:5: note:   conversion of argument 1 would be ill-formed:
In file included from ../../Tests/LibJS/test-js.cpp:8:
../../Userland/Libraries/LibTest/JavaScriptTestRunner.h:210:21: error: invalid conversion from ‘const char*’ to ‘u32’ {aka ‘unsigned int’} [-fpermissive]
  210 |     define_property("global", this, JS::Attribute::Enumerable);
      |                     ^~~~~~~~
      |                     |
      |                     const char*
../../Userland/Libraries/LibTest/JavaScriptTestRunner.h:210:21: error: invalid conversion from ‘const char*’ to ‘u32’ {aka ‘unsigned int’} [-fpermissive]
In file included from ../../Userland/Libraries/LibJS/Runtime/Object.h:17,
                 from ../../Userland/Libraries/LibJS/Runtime/EnvironmentRecord.h:9,
                 from ../../Userland/Libraries/LibJS/Bytecode/Op.h:17,
                 from ../../Userland/Libraries/LibJS/Bytecode/Generator.h:14,
                 from ../../Userland/Libraries/LibJS/Bytecode/Interpreter.h:9,
                 from ../../Userland/Libraries/LibTest/JavaScriptTestRunner.h:22,
                 from ../../Tests/LibJS/test-js.cpp:8:
../../Userland/Libraries/LibJS/Runtime/PropertyName.h:44:22: note:   initializing argument 1 of ‘JS::PropertyName::PropertyName(u32)’
   44 |     PropertyName(u32 index)
      |                  ~~~~^~~~~
In file included from ../../Userland/Libraries/LibJS/Runtime/EnvironmentRecord.h:9,
                 from ../../Userland/Libraries/LibJS/Bytecode/Op.h:17,
                 from ../../Userland/Libraries/LibJS/Bytecode/Generator.h:14,
                 from ../../Userland/Libraries/LibJS/Bytecode/Interpreter.h:9,
                 from ../../Userland/Libraries/LibTest/JavaScriptTestRunner.h:22,
                 from ../../Tests/LibJS/test-js.cpp:8:
../../Userland/Libraries/LibJS/Runtime/Object.h:92:26: note:   initializing argument 1 of ‘bool JS::Object::define_property(const JS::PropertyName&, JS::Value, JS::PropertyAttributes, bool)’
   92 |     bool define_property(const PropertyName&, Value value, PropertyAttributes attributes = default_attributes, bool throw_exceptions = true);
      |                          ^~~~~~~~~~~~~~~~~~~
```

Reply:
Okay, this makes absolutely no sense.

Reply:
Dropped the commit as discussed on Discord. :+1: